src/neo/SmartContract/ApplicationEngine.Storage.cs

```diff
@@ -104,32 +93,28 @@ private void PutExInternal(StorageContext context, byte[] key, byte[] value, Sto
             }
             else
             {
-                if (item.IsConstant) throw new InvalidOperationException();
                 if (value.Length == 0)
-                    newDataSize = 1;
+                    newDataSize = 0;
                 else if (value.Length <= item.Value.Length)
                     newDataSize = (value.Length - 1) / 4 + 1;
+                else if (item.Value.Length == 0)
+                    newDataSize = value.Length;
                 else
                     newDataSize = (item.Value.Length - 1) / 4 + 1 + value.Length - item.Value.Length;
             }
             AddGas(newDataSize * StoragePrice);
 
             item.Value = value;
-            item.IsConstant = flags.HasFlag(StorageFlags.Constant);
         }
 
         protected internal void Delete(StorageContext context, byte[] key)
         {
             if (context.IsReadOnly) throw new ArgumentException();
-            AddGas(StoragePrice);
```

Comment:
Why remove this line?

Reply:
I've added fee in `Register`.

Reply:
But this will be fixed, not defined by policy, but it's fine by me.